cameo/strain_design/deterministic/linear_programming.py

```diff
@@ -426,3 +430,394 @@ def _repr_html_(self):
         </ul>
         %s""" % (self._target, self.data_frame._repr_html_())
         return html_string
+
+
+class GrowthCouplingPotential(StrainDesignMethod):
+    """
+    Optimize the Growth Coupling Potential through a combination of knockouts, knockins and medium additions
+
+    Use the 'run' method to optimize the MILP problem and find the best solution.
+
+    In order to find multiple sub-optimal solutions through the use of solution pools, please refer to the
+    relevant solvers documentation. The underlying solver object can be accessed through
+        GrowthCouplingPotential(...).model.solver.problem
+
+
+    Parameters
+    ----------
+    model : cobra.Model
+        A cobra model object containing the target reaction as well as all native and heterologous reactions.
+    target : cobra.Reaction or string
```

Comment:
This differ from the signature of OptKnock class and its run method: 
```
OptKnock(model, exclude_reactions=None, remove_blocked=True, fraction_of_optimum=0.1, exclude_non_gene_reactions=True, use_nullspace_simplification=True, *args, **kwargs)
run(self, max_knockouts=5, biomass=None, target=None, max_results=1, *args, **kwargs)
```
An OptKnock object is created without mention of the target, but it is a parameter set in the `run()` method. Same remark for the number of knockouts. I suggest that it should be harmonized one way or the other in both classes, I guess keeping OptKnock strategy for backward compatibility.

Reply:
Thanks for pointing that out. Regarding the number of knockouts etc it should be relatively easy to change to follow the OptKnock signature. The target is a bit more tricky. The reason I put it as an argument to `__init__` is that the choice of target reaction affects the construction of the optimization problem more than is the case for OptKnock. I will think about whether/how this can be changed

Reply:
Honestly, it might be better for cameo to accept that the signature between optimisation classes (and there associated run method)  is **expected** to be different for each class. This would give more flexibility to create the best preprocessing environment for the problem, just like you did. However this is far from an "object-oriented" mindset, which can be surprising for users. So, an acceptable solution IMHO would be to simply warn users in cameo documentation, but really, this is a design decision that should be made by package maintainers since the question will arise for each new algorithm implemented in cameo.